app/interactors/access_limit/update_interactor.rb

```diff
@@ -22,12 +22,19 @@ def find_and_lock_edition
   end
 
   def update_edition
-    context.fail! if edition.access_limit&.limit_type == params[:limit_type]
-    edition.assign_access_limit(params[:limit_type], user).save!
+    if params[:limit_type] == "none"
+      context.fail! if edition.access_limit.nil?
+      edition.remove_access_limit(user).save!
+    else
+      context.fail! if edition.access_limit&.limit_type == params[:limit_type]
+      edition.assign_access_limit(params[:limit_type], user).save!
```

Comment:
Only thing of note here, that I didn't notice last time around, is that an unexpected limit_type value would cause app to exception quite far down the stack.

Reply:
True. The message from ActiveRecord seems helpful enough, so I don't think we don't need to check it again.

Reply:
You know me, I don't like that an error with client input causes an internal server error rather than a client error.

Reply:
It's good to think about it. Looks like it's hard to do this with params, but I've gone with a compromise to give a bad request if the param isn't there.

Reply:
Sure, I was expecting this to be something thought about requirements point since we could check it's there and an expected value in one check.

Reply:
Not sure I follow there - what would we do if the check failed? It doesn't sound like a requirements issue to me.

Reply:
I was thinking of it in the same vein as: https://github.com/alphagov/content-publisher/blob/master/app/interactors/publish/publish_interactor.rb#L37-L39

Reply:
Still a bit confused :-). Let's talk in person. Anyway, it looks like this particular change is good to merge, so I'll get it in now.